### PR TITLE
Update ComWrappers.xml

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
## Summary

Provide some technical details about the usage/expectations of the `ComWrappers` API.

/cc @jkoritzinsky @elinor-fung @scottj1s

### CODE REVIEW

File: xml/System.Runtime.InteropServices/ComWrappers.xml [hunk 1]
```diff
@@ -25,7 +25,59 @@
...
+|   Manage object    |                |     Native proxy     |
+|                    |                | Ref count: 1         |
+|  ----------------  |                |  ------------------  |
+| | Weak reference |=| = = = = = = = >| | Strong reference | |
+| |    to proxy    | |<===============|=|    to object     | |
+|  ----------------  |                |  ------------------  |
+ --------------------                  ----------------------
+```
+
+The state of the native object and  managed proxy after a call to [`ComWrappers.GetOrCreateObjectForComInstance`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.comwrappers.getorcreateobjectforcominstance) is illustrated below. The concept of "identity" follows the [rules for `IUnknown`](https://docs.microsoft.com/windows/win32/com/rules-for-implementing-queryinterface#objects-must-have-identity).
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/ComWrappers.xml:62]:
```suggestion
The next illustration shows the state of the native object and managed proxy after a call to <xref:System.Runtime.InteropServices.ComWrappers.GetOrCreateObjectForComInstance(System.IntPtr,System.Runtime.InteropServices.CreateObjectFlags)?displayProperty=nameWithType>. The concept of "identity" follows the [rules for `IUnknown`](https://docs.microsoft.com/windows/win32/com/rules-for-implementing-queryinterface#objects-must-have-identity).
```

File: xml/System.Runtime.InteropServices/ComWrappers.xml [hunk 2]
```diff
@@ -25,7 +25,59 @@
...
+ ------------------              | native identity  |
+ ------------------------        | to managed proxy |
+|   Managed proxy        |< = = =|                  |
+| Created by ComWrappers |        ------------------
+|   implementer.         |
+| Optional AddRef() on   |
+|   native object.       |
+ ------------------------
+```
+Observe in the above illustration only weak references exist from the runtime perspective. The `+1` reference count on the native object is assumed to be performed by the managed proxy creator (i.e. `ComWrappers` implementer) to ensure the associated lifetime between the native object and its managed proxy. There is an optional strong reference (i.e. `AddRef()`) mentioned in the managed proxy, which is used to support scenario (3) above - see [`CreateObjectFlags.TrackerObject`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.createobjectflags). With this optional strong reference the reference count would be `+2`.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/ComWrappers.xml:77]:
```suggestion
Observe that only weak references exist from the runtime perspective. The `+1` reference count on the native object is assumed to be performed by the managed proxy creator (that is, the `ComWrappers` implementer) to ensure the associated lifetime between the native object and its managed proxy. There is an optional strong reference (that is, `AddRef()`) mentioned in the managed proxy, which is used to support scenario (3) mentioned earlier. See <xref:System.Runtime.InteropServices.CreateObjectFlags.TrackerObject?displayProperty=nameWithType>. With this optional strong reference, the reference count would be `+2`.
```

**Comment by AaronRobinsonMSFT** [comment]
@gewarren Thanks for the feedback. I am still working through some additional details to add, so should I convert this to draft?

**Comment by gewarren** [comment]
> @gewarren Thanks for the feedback. I am still working through some additional details to add, so should I convert this to draft?

That's probably a good idea.

**Comment by AaronRobinsonMSFT** [comment]
@gewarren I believe this is indeed good from my end now. However, there appears to be a link issue to the `IReferenceTracker` API. What do I need to do in order to mitigate that?

**Comment by gewarren** [comment]
> @gewarren I believe this is indeed good from my end now. However, there appears to be a link issue to the `IReferenceTracker` API. What do I need to do in order to mitigate that?

I'm not sure. @joelmartinez @TianqiZhang is this a case where we need to add the link to missingapis.xml?

**Comment by joelmartinez** [comment]
@gewarren I'm not familiar with the `NN:` prefix on that xref (I guess it's a native API?). But in any case, yes if you add that docid to the missing apis file, then it will resolve correctly.

File: xml/System.Runtime.InteropServices/ComWrappers.xml [hunk 3]
```diff
@@ -25,7 +25,61 @@
...
+
+Traditionally in the runtime, a native proxy to managed object is called a COM Callable Wrapper (CCW), and a managed proxy to a native object is called a Runtime Callable Wrapper (RCW). However, when used here, those terms should not be confused with the built-in features of the same name (that is, [CCW](https://docs.microsoft.com/dotnet/standard/native-interop/com-callable-wrapper) and [RCW](https://docs.microsoft.com/dotnet/standard/native-interop/runtime-callable-wrapper)). Unlike the built-in features, a majority of the responsibility for accurate lifetime management, dispatching methods, and marshalling of arguments and return values is left to the `ComWrappers` implementer.
+
+"Minimal support" is defined by the following features:
+
+1. Efficient mapping between a managed object and a native proxy (for example, CCW).
+2. Efficient mapping between a native `IUnknown` and its managed proxy (for example, RCW).
+3. Integration with the garbage collector through the <xref:NN:windows.ui.xaml.hosting.referencetracker.IReferenceTrackerHost> interface contract.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/ComWrappers.xml:41]:
```suggestion
3. Integration with the garbage collector through the [IReferenceTrackerHost](/windows/win32/api/windows.ui.xaml.hosting.referencetracker/nn-windows-ui-xaml-hosting-referencetracker-ireferencetrackerhost) interface contract.
```

Sorry, I think I'm the one who made it an xref in the first place.

**Review by gewarren — APPROVED** [review]
I verified the link to IReferenceTrackerHost works, so I think this is ready to be merged.

**Pull request merged**